docs/docs/usage/hooks.md

```diff
@@ -0,0 +1,234 @@
+
+- [`post_init`](../plugin/reference.md#pdm.signals.post_init)
+
+``` mermaid
+flowchart LR
+  subgraph pdm-new [pdm new]
+    direction LR
+
+    subgraph pdm-init [pdm init]
+      direction LR
+      post-init{{Emit post_init}}
+      init --> post-init
+    end
+    
+    new --> pdm-init
+  end
+```
+
+## Dependencies management
+
+The dependencies management is required for the developper to be able to work and perform the following:
+
+- `lock`: compute a lock file from the `pyproject.toml` requirements
+- `sync`: synchronize (add/remove/update) PEP582 packages from the lock file
+- `install`: install the current project as editable
+- `add`: add a dependency
+- `remove`: remove a dependency
+
+All thoses steps are directly available with the following commands:
+
+- [`pdm lock`](dependency.md#install-the-packages-pinned-in-lock-file): execute the `lock` task
+- [`pdm sync`](dependency.md#install-the-packages-pinned-in-lock-file): execute the `sync` task
+- [`pdm install`](dependency.md#install-the-packages-pinned-in-lock-file): execute the `install` task, preceded from `lock` and/or `sync` if required
+- [`pdm add`](dependency.md#add-dependencies): add a dependency requirement, re-lock and then sync
+- [`pdm remove`](dependency.md#remove-existing-dependencies): remove a dependency requirement, re-lock and then sync
+- [`pdm update`](dependency.md#update-existing-dependencies): re-lock dependencies from their latest versions and then sync
+
+They trigger the following hooks:
+
+- [`pre_install`](../plugin/reference.md#pdm.signals.pre_install)
+- [`post_install`](../plugin/reference.md#pdm.signals.post_install)
+- [`pre_lock`](../plugin/reference.md#pdm.signals.pre_lock)
+- [`post_lock`](../plugin/reference.md#pdm.signals.post_lock)
+- [`pre_sync`](../plugin/reference.md#pdm.signals.pre_sync)
+- [`post_sync`](../plugin/reference.md#pdm.signals.post_sync)
+
+
+``` mermaid
+flowchart LR
+  subgraph pdm-install [pdm install]
+    direction LR
+    pre-install{{Emit pre_install}}
+    post-install{{Emit post_install}}
+
+    subgraph pdm-lock [pdm lock]
+      direction TB
+      pre-lock{{Emit pre_lock}}
+      post-lock{{Emit post_lock}}
+      pre-lock --> lock --> post-lock
+    end
+    
+    subgraph pdm-sync [pdm sync]
+      direction TB
+      pre-sync{{Emit pre_sync}}
+      post-sync{{Emit post_sync}}
+      pre-sync --> sync --> post-sync
+    end
+    
+    pre-install --> pdm-lock --> pdm-sync --> install --> post-install
+  end
+```
+
+## Publication
+
+As soon as you are ready to publish you package/app/libray, you will required the publish flow:
+
+- `build`: build/compile assets requiring it
+- `package`: package you library/application into a Python package
+- `upload`: upload/publish the package to a remote PyPI index
+
+All thoses steps are available with the following commands:
+
+- [`pdm build`](project.md#build-distribution-artifacts)
+- [`pdm publish`](project/#build-distribution-artifacts)
```

Comment:
This made me notice a discrepancy: `(page.md#...)` versus `(page/#...)`. I find the latter more readable.

Reply:
But this version was not working, it was missing the start of the path.
If finally chose the first one because of it: `.md` links are resolved by mkdocs and so survive URL changes and support proper sibling files references , while the second one forced you to know the expected final path.